parser/encoding/request.go

```diff
@@ -0,0 +1,138 @@
+package encoding
+
+import (
+	"github.com/cockroachdb/errors"
+
+	"encr.dev/parser"
+	v1 "encr.dev/proto/encore/parser/schema/v1"
+)
+
+// ParamLocation is the request/response home of the parameter
+type ParamLocation int16
+
+const (
+	Header = 1 // Parameter is placed in the http header
+	Query  = 2 // Parameter is placed in the query string
+	Body   = 3 // Parameter is placed in the body
+)
+
+// tagToLocation is used to translate struct field tags to param locations
+// if overrideDefault is set, the default http param location will be changed
+// if the tag matches the paramLocation, the param name will be override with the
+// tag name
+var tagToLocation = map[string]struct {
+	location        ParamLocation
+	overrideDefault bool
+}{
+	"query": {
+		location:        Query,
+		overrideDefault: true,
+	},
+	"qs": {
+		location:        Query,
+		overrideDefault: true,
+	},
+	"header": {
+		location:        Header,
+		overrideDefault: true,
+	},
+	"json": {
+		location:        Body,
+		overrideDefault: false,
+	},
+}
+
+// RequestEncoding expresses how a request should be encoded for a specified set of HttpMethods
+type RequestEncoding struct {
+	// The HTTP methods these field configurations can be used for
+	HttpMethods []string
+	// Contains metadata about how to marshal a http parameter
+	Fields []*ParameterEncoding
+}
+
+// ParameterEncoding expresses how a parameter should be encoded
+type ParameterEncoding struct {
+	// Location (e.g. header, query, body) where the parameter should be marshalled from/to
+	Location ParamLocation
```

Comment:
A bit strange to see `ParameterEncoding` but `ParamLocation`. Probably `Param` is clear enough that we should use that instead of `Parameter` everywhere?